src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -24,28 +23,164 @@ class ThemePatch : ResourcePatch {
+             * @field filePathIndex The current styles.xml path, indexed by the method 'forEachIndexed()'
+             */
+            file.createElement("item").apply {
+                setAttribute(
+                    "name",
+                    when (filePathIndex) {
+                        0 -> splashScreenColorAttributeName
+                        1 -> "android:windowSplashScreenBackground"
+                        else -> "null"
+                    }
+                )
 
-                node.textContent = when (node.getAttribute("name")) {
-                    "yt_black0", "yt_black1", "yt_black1_opacity95", "yt_black1_opacity98", "yt_black2", "yt_black3",
-                    "yt_black4", "yt_status_bar_background_dark", "material_grey_850" -> darkThemeBackgroundColor
+                /**
+                 * Fill the value of created node element with the field 'lightThemeBackgroundColor' or 'darkThemeBackgroundColor'
+                 * only if the name of 'indexedNode' is equal to the field 'nodeAttributeName'
+                 */
+                appendChild(
+                    file.createTextNode(
+                        when (filePathIndex) {
+                            0 -> when (nodeAttributeName) {
+                                "Base.Theme.YouTube.Launcher.Dark" -> darkThemeBackgroundColor
+                                "Base.Theme.YouTube.Launcher.Light" -> lightThemeBackgroundColor
+                                else -> "null"
+                            }
+                            1 -> when (nodeAttributeName) {
+                                "Base.Theme.YouTube.Launcher" -> "?attr/$splashScreenColorAttributeName"
+                                else -> "null"
+                            }
+                            else -> "null"
+                        }
+                    )
+                )
 
-                    "yt_white1", "yt_white1_opacity95", "yt_white1_opacity98", "yt_white2", "yt_white3", "yt_white4",
-                     -> lightThemeBackgroundColor
+                /**
+                 * Add the created node element after 'indexedNode', into xml file
+                 * only if the value of new node element is not equal to 'null'
+                 */
+                if (this.textContent != "null")
+                    indexedNode.appendChild(this)
+            }
+        }
+        fun changeSplashScreenBackgroundColor(rootNode: Element) {
+            /**
+             * Check if a node element with attribute 'android:drawable' exists
+             * then change its value to '?attr/splashScreenColor'
+             */
+            if (rootNode.attributes.getNamedItem("android:drawable") != null)
+                rootNode.setAttribute("android:drawable", "?attr/splashScreenColor")
+        }
+
+        fun editXMLFile(
+            filePath: String,
+            filePathIndex: Int,
+            elementsToSearchName: String,
+            patchIndex: Int
+        ) {
```

Comment:
This method's signature is unreasonable. What is `filePathIndex` or `patchIndex`? What does it have to do with `editXMLFile` as the method name conveys?

Reply:
```filePathIndex``` it is used to apply two distinct patches to the style.xml files.

```patchIndex``` it is used to choose which patch to apply.

But already there are comments that explain it.

Reply:
This is too arbitrary, simplify and split your method reasonably.